@here/harp-mapview/lib/VisibleTileSet.ts

```diff
@@ -246,78 +313,96 @@ export class VisibleTileSet {
                 }
 
-                if (area <= 0 || tileEntry.tileKey.level > displayZoomLevel) {
+                if (area <= 0 || tileKey.level > displayZoomLevel) {
                     continue;
                 }
 
-                if (dataSource.shouldRender(displayZoomLevel, tileEntry.tileKey)) {
+                if (dataSource.shouldRender(displayZoomLevel, tileKey)) {
                     visibleTiles.push(tileEntry);
                 }
 
-                tilingScheme.getSubTileKeys(tileEntry.tileKey).forEach(childTileKey => {
-                    const intersectsFrustum = tileFrustumIntersectionCache.get(
-                        childTileKey.mortonCode()
-                    );
-
-                    let subTileArea = 0;
-
-                    if (intersectsFrustum === undefined) {
-                        const geoBox = tilingScheme.getGeoBox(childTileKey);
-                        this.options.projection.projectBox(geoBox, tileBounds);
-                        tileBounds.min.sub(worldCenter);
-                        tileBounds.max.sub(worldCenter);
-
-                        if (
-                            (!this.options.extendedFrustumCulling ||
-                                this.m_mapTileCuller.frustumIntersectsTileBox(tileBounds)) &&
-                            this.m_frustum.intersectsBox(tileBounds)
-                        ) {
-                            const contour = [
-                                new THREE.Vector3(
-                                    tileBounds.min.x,
-                                    tileBounds.min.y,
-                                    0
-                                ).applyMatrix4(this.m_viewProjectionMatrix),
-                                new THREE.Vector3(
-                                    tileBounds.max.x,
-                                    tileBounds.min.y,
-                                    0
-                                ).applyMatrix4(this.m_viewProjectionMatrix),
-                                new THREE.Vector3(
-                                    tileBounds.max.x,
-                                    tileBounds.max.y,
-                                    0
-                                ).applyMatrix4(this.m_viewProjectionMatrix),
-                                new THREE.Vector3(
-                                    tileBounds.min.x,
-                                    tileBounds.max.y,
-                                    0
-                                ).applyMatrix4(this.m_viewProjectionMatrix)
-                            ];
-
-                            contour.push(contour[0]);
-
-                            const n = contour.length;
-
-                            for (let p = n - 1, q = 0; q < n; p = q++) {
-                                subTileArea +=
-                                    contour[p].x * contour[q].y - contour[q].x * contour[p].y;
+                tilingScheme.getSubTileKeys(tileKey).forEach(childTileKey => {
```

Comment:
Please add a comments that motivate the changes due to the use case

Reply:
Yes, this is required, I will move this into a separate function I think to make it clear

Reply:
Still open